src/formatting_stack/formatters/no_extra_blank_lines.clj

```diff
@@ -21,6 +22,29 @@
                                    (spit filename formatted))))))
   nil)
 
+(defn lint! [this files]
+  (->> files
+       (process-in-parallel! (fn [filename]
```

Comment:
This logic could be made pure by decoupling it from `filename`. That way, you could unit-test it, and extensively.

It'd be particularly important that formatting and linting tests shared the same exact corpus in `unit.formatting-stack.formatters.no-extra-blank-lines`,  that way we prove they behave consistently between them.